mesheryctl/internal/cli/root/system/channel.go

```diff
@@ -81,11 +82,37 @@ var viewCmd = &cobra.Command{
 	},
 }
 
+//check release channel string supplied by user for validity
+func validChannel(c string) bool {
+	switch c {
+	case "stable":
+		return true
+	case "edge":
+		return true
+	}
+	stableRegx := regexp.MustCompile(`^stable-v[0-9]+\.[0-9]+\.[0-9]+$`)
+	edgeRegx := regexp.MustCompile(`^edge-v[0-9]+\.[0-9]+\.[0-9]+$`)
+	return stableRegx.MatchString(c) || edgeRegx.MatchString(c)
```

Comment:
Although I love regex 😁, wouldn't it be cleaner if we just use the valid version check function? Now that we have eliminated the GH API calls, it wouldn't be much of an issue as well.

Reply:
You mean func IsBetaOrStable ? I hadn't seen it. I will use that instead of the switch. Are you refering to  ValidateVersion method of *Context object, Or I'm missing the something here.

Reply:
Yes, the validate version method of the context object. Basically, instead of regex, where the user could specify versions that are not yet released but are still valid, we can scrape GitHub to see if the version actually exists.

Reply:
recently Lee asked why are we including versions in channel?? Channel should only have "edge" or "stable", we have a separate version field. So I and prolly Lee doesn't get the point of including a version in the channel field?

Reply:
@piyushsingariya  The current behaviour is when no version is specified, it defaults to latest version. but the user may want to specify a version different from latest of the chosen channel. We can scrape github for valid channels and validate the version entered here.

Reply:
Yes. @Zanda256 That sounds right. Could you please make the changes as suggested here and we can merge this.